HHS/Head-Start-TTADP
# [TTAHUB-2259] Similarity node communication layer

## Description of change

This is the communication layer between node and the similarity API. A PR from @thewatermethod was merged into this that formats the response for consumption by the FE.

One route is added:

```
GET /goals/similar/:recipientId?cluster=true/false
```

## How to test

Make sure you have a value for `SIMILARITY_ENDPOINT` in your `.env`. Locally, this value should be `http://similarity_api:9100/compute`. Go to the RTR, select a recipient and go to the RTTAPA page. Look in the network page and ensure the call to the endpoint was successful and returns something like:

```javascript
[
  {
    "ids": [...],
    "goals": [
      {
        "name": "...",
        "source": null,
        "status": "Closed",
        "responsesForComparison": "",
        "ids": [...]
      },
      {
        "name": "...",
        "source": null,
        "status": "Closed",
        "responsesForComparison": "",
        "ids": [...]
      }
    ]
  }
]
```



## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-2259


## Checklists

### Every PR



### Production Deploy


### After merge/deploy
